manual/src/ornate/reference.md

```diff
@@ -64,24 +64,94 @@ useYarn := true
+}
+<script language="JavaScript">
+var exports = window;
+exports.require = window["ScalaJsBundlerDependencies"].require;
+</script>
+@defining(s"${projectName.toLowerCase}") { name =>
+  @Seq(s"$name-opt.js", s"$name-fastopt.js").find(resourceExists).map(name => jsScript(assets(name), htmlAttributes))
+}
+~~~
+
+The default variable global for the library is `ScalaJsBundlerDependencies`. Should you need to change it, 
+you can provide a new variable name to the configuration, such as `bundlingMode := BundlingMode.LibrariesOnly("myLib")`.
+
+##### Benefits
+
+By avoiding processing the entire scala.js output file, webpack times from 10's of seconds to minutes to 
+a couple of seconds, depending on how many dependencies webpack has to resolve. In addition, since the 
+module references are all still managed by webpack, any custom loaders such as the `html-loader` or `text-loader`
+will work as usual. Other webpack configuration such as external modules also work out of the box.
+
+By splitting the output into a separate library and scala.js app we are able to leverage browser caching for most
+reloads, since it's seldom that both .js assets and scala.js assets change at the same time. 
+
+By avoiding any post-processing of the scala.js output, we leave the sourcemap generated by the scala.js compiler
+entirely intact. This means we avoid any translation or lookup errors introduced by webpack, and also means
+we save a bunch more processing time even if sourcemaps are enabled.
+
+##### How It Works
+Scalajs-bundler has a unique advantage in the JavaScript build ecosystem in that it runs inside SBT and has
+access to the Scala.Js compiler internals. By leveraging these, we can generate a fake `entrypoint.js` file that 
+represents all the `@JSImport`s from a Scala.JS project. We can use this entrypoint to generate a library file
+that provides access to all those imports without webpack having to ever process the entire scala.js output. 
+
+Inside the `entrypoint.js` file we also provide a trivial `require` implementation that provides access to the
+modules which were imported.
+
+All that remains is to provide an `exports` variable with a reference to the `require` implementatino 
+for the scala.js out to hang onto, which is what the loader provides.
+
+#### Library and Application {#bundling-mode-library-and-application}
+
+`bundlingMode := BundlingMode.LibraryAndApplication()` attempts to duplicate the output of `BundlingMode.Application` without
+the overhead. It does this by concatenating the library, loader, and scala.js output into a `yourapp-bundle` 
+file.
+
+If you need interchangability with the full `Application` mode or somehow can't handle the two files generated
+by the `LibraryOnly()` mode, this mode may be useful for you. However, it relies on post-processing and eliminates
+some of the benefits of the `LibraryOnly` mode, so consider carefully if you really need it before turning it on.
```

Comment:
You should also mention that this mode also makes use of the `output.library` Webpack setting